### PR TITLE
Fix embedded package panic

### LINKED ISSUE #4235: batch-size error when using embbeded ferretdb package
We should set that field for the embeddable package there: https://github.com/FerretDB/FerretDB/blob/6cda9ad89a0bcd7aac6c8bb4cba65b11532fe6d0/ferretdb/ferretdb.go#L137-L139

---

### FerretDB version

0.21.0

### Backend

SQLite

### Environment

- OS: Windows 11
- Deployment: Embbeded
- Deployment details:

### What did you do?

Insert document

```
package main

import (
	"context"
	"log"

	"github.com/FerretDB/FerretDB/ferretdb"
	"go.mongodb.org/mongo-driver/bson"
	"go.mongodb.org/mongo-driver/mongo"
	"go.mongodb.org/mongo-driver/mongo/options"
)

func main() {
	// start server
	f, err := ferretdb.New(&ferretdb.Config{
		Listener: ferretdb.ListenerConfig{
			TCP: "127.0.0.1:0",
		},
		Handler:   "sqlite",
		SQLiteURL: "file:./?mode=memory",
	})
	if err != nil {
		log.Fatal(err)
	}

	ctx, cancel := context.WithCancel(context.Background())

	done := make(chan struct{})

	go func() {
		log.Print(f.Run(ctx))
		close(done)
	}()

	uri := f.MongoDBURI()

	// start client
	mongoCli, _ := mongo.Connect(ctx, options.Client().ApplyURI(uri))
	_, e := mongoCli.Database("test").Collection("test").InsertOne(context.TODO(), bson.M{"abc": "def"})
	if e != nil {
		log.Print(e)
	}
	cancel()
	<-done
}
```

### What did you expect to see?

No error

### What did you see instead?

```
2024-04-15T15:59:25.084+0800    DPANIC  // 127.0.0.1:52039 -> 127.0.0.1:52037   clientconn/conn.go:170  batch-size should be greater or equal to 1
(err = <nil>)
github.com/FerretDB/FerretDB/internal/clientconn.(*conn).run.func3
        C:/Users/ben/go/pkg/mod/github.com/!ferret!d!b/!ferret!d!b@v1.21.0/internal/clientconn/conn.go:170
runtime.gopanic
        C:/Users/ben/scoop/apps/go/current/src/runtime/panic.go:770
github.com/FerretDB/FerretDB/internal/backends/sqlite.(*collection).InsertAll.func1
        C:/Users/ben/go/pkg/mod/github.com/!ferret!d!b/!ferret!d!b@v1.21.0/internal/backends/sqlite/collection.go:114
github.com/FerretDB/FerretDB/internal/util/fsql.(*DB).InTransaction
        C:/Users/ben/go/pkg/mod/github.com/!ferret!d!b/!ferret!d!b@v1.21.0/internal/util/fsql/db.go:172
github.com/FerretDB/FerretDB/internal/backends/sqlite.(*collection).InsertAll
        C:/Users/ben/go/pkg/mod/github.com/!ferret!d!b/!ferret!d!b@v1.21.0/internal/backends/sqlite/collection.go:111
github.com/FerretDB/FerretDB/internal/backends.(*collectionContract).InsertAll
        C:/Users/ben/go/pkg/mod/github.com/!ferret!d!b/!ferret!d!b@v1.21.0/internal/backends/collection.go:202
github.com/FerretDB/FerretDB/internal/backends/decorators/oplog.(*collection).InsertAll
        C:/Users/ben/go/pkg/mod/github.com/!ferret!d!b/!ferret!d!b@v1.21.0/internal/backends/decorators/oplog/collection.go:64
github.com/FerretDB/FerretDB/internal/handler.(*Handler).MsgInsert
        C:/Users/ben/go/pkg/mod/github.com/!ferret!d!b/!ferret!d!b@v1.21.0/internal/handler/msg_insert.go:145
github.com/FerretDB/FerretDB/internal/handler.(*Handler).initCommands.(*Handler).authenticateWrapper.func1
        C:/Users/ben/go/pkg/mod/github.com/!ferret!d!b/!ferret!d!b@v1.21.0/internal/handler/commands.go:304
github.com/FerretDB/FerretDB/internal/clientconn.(*conn).handleOpMsg
        C:/Users/ben/go/pkg/mod/github.com/!ferret!d!b/!ferret!d!b@v1.21.0/internal/clientconn/conn.go:575
github.com/FerretDB/FerretDB/internal/clientconn.(*conn).route
        C:/Users/ben/go/pkg/mod/github.com/!ferret!d!b/!ferret!d!b@v1.21.0/internal/clientconn/conn.go:432
github.com/FerretDB/FerretDB/internal/clientconn.(*conn).run
        C:/Users/ben/go/pkg/mod/github.com/!ferret!d!b/!ferret!d!b@v1.21.0/internal/clientconn/conn.go:316
github.com/FerretDB/FerretDB/internal/clientconn.acceptLoop.func1
        C:/Users/ben/go/pkg/mod/github.com/!ferret!d!b/!ferret!d!b@v1.21.0/internal/clientconn/listener.go:305
```

**Additional**

0.20.1 does not have such error.

### PR DESCRIPTION
**Pull request by noisersup**
# Description

Closes #4235.

## Readiness checklist

- [ ] I added/updated unit tests (and they pass).
- [ ] I added/updated integration/compatibility tests (and they pass).
- [ ] I added/updated comments and checked rendering.
- [ ] I made spot refactorings.
- [ ] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

File: ferretdb/ferretdb.go [hunk 1]
```diff
@@ -134,8 +134,10 @@ func New(config *Config) (*FerretDB, error) {
 
 		SQLiteURL: config.SQLiteURL,
 
+		//nolint:mnd // Handler default values
```
Comment (noisersup) [inline ferretdb/ferretdb.go:137]: Those default values are handled in kong tags, hence I believe there's no need of introducing constants to remove complaints about magic numbers.

**Review by chilagrow — APPROVED** [review]

**Review by AlekSi — COMMENTED** [review]
We must add tests for fixed bugs reported by users

File: ferretdb/ferretdb.go (hunk 1 above)
Comment (AlekSi) [reply ferretdb/ferretdb.go:137]: > Those default values are handled in kong tags

So, they are not handler default values. The code comment is therefore incorrect

**Review by rumyantseva — COMMENTED** [review]
The change looks good, even better if we can add tests, yes :)

File: ferretdb/ferretdb_test.go [hunk 2]
```diff
@@ -139,3 +145,34 @@ func Example_tls() {
...
 	// Output: mongodb://127.0.0.1:17028/?tls=true
 }
+
+func TestFerretDB(t *testing.T) {
+	f, err := ferretdb.New(&ferretdb.Config{
+		Listener: ferretdb.ListenerConfig{
+			TCP: "127.0.0.1:0",
+		},
+		Handler:   "sqlite",
+		SQLiteURL: "file:./?mode=memory",
```
Comment (AlekSi) [inline ferretdb/ferretdb_test.go:155]: Let's keep that unit test, but make it use PostgreSQL instead.
And let's remove that integration test that we skip anyway.

File: ferretdb/ferretdb_test.go [hunk 3]
```diff
@@ -139,3 +145,34 @@ func Example_tls() {
...
+		require.NoError(t, f.Run(ctx))
+		close(done)
+	}()
+
+	uri := f.MongoDBURI()
+
+	client, err := mongo.Connect(ctx, options.Client().ApplyURI(uri))
+	require.NoError(t, err)
+
+	_, err = client.Database("test").Collection("test").InsertOne(ctx, bson.M{"foo": "bar"})
```
Comment (AlekSi) [inline ferretdb/ferretdb_test.go:173]: Let's just add `//nolint`

**Review by AlekSi — APPROVED** [review]

**Review by rumyantseva — APPROVED** [review]
Looks good!

**Pull request merged**